File: benchmarks/pandas/bench_math_ops.py
"""Benchmark: math_ops — abs / round on Series and DataFrame of 100k rows."""
import json, time
import numpy as np
import pandas as pd

SIZE = 100_000
WARMUP = 5
ITERATIONS = 50

s = pd.Series(np.where(np.arange(SIZE) % 2 == 0, -(np.arange(SIZE) + 0.567), np.arange(SIZE) + 0.567))
df = pd.DataFrame({
    "a": -(np.arange(SIZE) + 0.123),
    "b": np.arange(SIZE) + 0.456,
})

for _ in range(WARMUP):
    s.abs()
    df.abs()
    s.round(1)
    df.round(1)

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.abs()
    df.abs()
    s.round(1)
    df.round(1)
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "math_ops",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))

File: benchmarks/tsb/bench_math_ops.ts
/**
 * Benchmark: math_ops — absSeries / absDataFrame / roundSeries / roundDataFrame on 100k rows.
 * Outputs JSON: {"function": "math_ops", "mean_ms": ..., "iterations": ..., "total_ms": ...}
 */
import { Series, DataFrame, absSeries, absDataFrame, roundSeries, roundDataFrame } from "../../src/index.ts";

const SIZE = 100_000;
const WARMUP = 5;
const ITERATIONS = 50;

const s = new Series({ data: Array.from({ length: SIZE }, (_, i) => (i % 2 === 0 ? -(i + 0.567) : i + 0.567)) });
const df = DataFrame.fromColumns({
  a: Array.from({ length: SIZE }, (_, i) => -(i + 0.123)),
  b: Array.from({ length: SIZE }, (_, i) => i + 0.456),
});

for (let i = 0; i < WARMUP; i++) {
  absSeries(s);
  absDataFrame(df);
  roundSeries(s, 1);
  roundDataFrame(df, 1);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  absSeries(s);
  absDataFrame(df);
  roundSeries(s, 1);
  roundDataFrame(df, 1);
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "math_ops",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
